.github/workflows/static_analysis.yml

```diff
@@ -0,0 +1,85 @@
+# contains non-functional tests, like checks on docs
+# and code style
+name: Static Analysis
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+env:
+  CONDA_ENV: test-env
+  GITHUB_ACTIONS: 'true'
+  PYTHON_VERSION: 3.8
+
+jobs:
+  test:
+    name: ${{ matrix.task }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - task: lint
+          - task: check-docs
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 5
+          submodules: false
+      - name: Setup and run tests
+        shell: bash
+        run: |
+          export TASK="${{ matrix.task }}"
+          export OS_NAME="linux"
+          export COMPILER="gcc"
```

Comment:
I think we can move these variables to `env` section of this config file closer to `PYTHON_VERSION` and other constants.

Reply:
fixed in https://github.com/microsoft/LightGBM/pull/3726/commits/1af56f4c9048084ab746c802a7957f350d3465d0